### PR TITLE
Transactions

### PR DESCRIPTION
**Pull request by badger-99**
## This PR will make the following alterations to the `.sql` files in this repo.
- The command to create the `animals` table in [schema.sql](https://github.com/badger-99/vet-clinic-database/blob/dev/schema.sql) is updated to include a _species_ column in the table.
- Additional commands to populate the table have been included in [data.sql](https://github.com/badger-99/vet-clinic-database/blob/dev/data.sql).
- Transactions and queries that involve aggregate functions and groupings have been added in [querries.sql](https://github.com/badger-99/vet-clinic-database/blob/dev/queries.sql).

<hr>

UPDATE ANIMALS SET species='digimon' WHERE name like '%mon';
UPDATE ANIMALS SET species='pokemon' WHERE species is null;
SELECT * FROM animals;
COMMIT;
SELECT * FROM animals;
Capture1

> BEGIN;
DELETE FROM animals;
ROLLBACK;
SELECT * FROM animals;
Capture2

> BEGIN;
DELETE FROM animals where date_of_birth>'2022-01-01';
SAVEPOINT point1;
UPDATE animals SET weight_kg=weight_kg*-1;
ROLLBACK TO point1;
UPDATE animals SET weight_kg=weight_kg*-1 WHERE weight_kg<0;
COMMIT;
SELECT * FROM animals ORDER BY id;
Capture3

Capture1

> SELECT COUNT(escape_attempts) AS no_escape_attempts FROM animals WHERE escape_attempts=0;
Capture2

>SELECT AVG(weight_kg) AS average_weight FROM animals;
Capture3

> SELECT neutered, SUM(escape_attempts) AS escape_Attempts
FROM animals GROUP BY neutered ORDER BY escape_attempts DESC;
Capture4

> SELECT species, MAX(weight_kg) AS maximum_weight_kg,
  MIN(weight_kg) AS minimum_weight_kg FROM animals GROUP BY species;
Capture5

> SELECT species, ROUND(AVG(escape_attempts), 2) AS escape_attempts
  FROM animals WHERE EXTRACT(YEAR FROM date_of_birth) BETWEEN 1990 AND 2000
  GROUP BY species;
Capture6

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -7,4 +7,51 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
...
+UPDATE ANIMALS SET species='unspecified';
+ROLLBACK;
+SELECT * FROM animals;
+
+-- Transaction 2
+BEGIN;
+UPDATE ANIMALS SET species='digimon' WHERE name like '%mon';
+UPDATE ANIMALS SET species='pokemon' WHERE species is null;
+COMMIT;
+SELECT * FROM animals;
```
Comment (AbelG101) [inline queries.sql]: - [ ] As part of the project requirements please make sure to  verify that change was made before committing the changes. This means you'll have to add a statement similar to the one at line `23`

File: queries.sql [hunk 2]
```diff
@@ -7,4 +7,51 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
...
 SELECT * FROM animals WHERE name NOT IN ('Gabumon');
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 and 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 and 17.3;
+
+-- Transaction 1
+BEGIN;
+UPDATE ANIMALS SET species='unspecified';
+ROLLBACK;
+SELECT * FROM animals;
```
Comment (AbelG101) [inline queries.sql]: - [ ] Once again please make sure to check the changes before rolling back the update.

File: queries.sql [hunk 3]
```diff
@@ -7,4 +7,51 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
...
+ROLLBACK TO point1;
+UPDATE animals SET weight_kg=weight_kg*-1 WHERE weight_kg<0;
+COMMIT;
+SELECT * FROM animals ORDER BY id;
+
+/*More Queries.*/
+SELECT COUNT(name) AS total_animals FROM animals;
+
+SELECT COUNT(escape_attempts) AS animals_tried_escaping
+  FROM animals WHERE escape_attempts>0;
```
Comment (AbelG101) [inline queries.sql]: - [ ] The question for this query asks for animals that have never tried to escape which means the `escape_attempts` should be equal to not greater than 0.

**Review by AbelG101 — CHANGES_REQUESTED** [review]
# Hi @badger-99 👋, 

You're doing a good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Species column is properly added ✔
- Proper transaction management ✔
- Correct use of Git Flow ✔
- Descriptive PR title with a good screenshot summary ✔

## Required Changes ♻️
- _Check the comments under the review._
- you just have to make a few changes and you're on your way 🚀.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @badger-99, 

To highlight 👍🏽 
✔️ Queries are successful
✔️ Screenshots are added

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**